clearnet/rpc.md

```diff
@@ -0,0 +1,263 @@
+  "type": "auth_success",
+  "data": {
+    "address": "0xAlice"
+  }
+}
+```
+
+---
+
+## 🛠️ Broker Methods
+
+### 1. `CreateVirtualChannel`
+
+#### ➤ Request
+```json
+{
+  "req": [1, "CreateVirtualChannel", [
+    {
+      "participants": ["0xAlice", "0xBob"],
+      "allocations": [
+        { "participant": "0xAlice", "token_address": "0xToken1", "amount": "1000000000000000000" },
+        { "participant": "0xBob",   "token_address": "0xToken1", "amount": "1000000000000000000" }
+      ],
+      "signers": ["0xAlice", "0xBob"] // Participants agree on a set of signers required to close the channel.
+    }
+  ], 1713870000],
+  "sig": ["sigAlice", "sigBob"] // Message should be signed by all participants with non-zero allocation.
+}
+```
+
+#### ⇦ Response
+```json
+{
+  "res": [1, "CreateVirtualChannel", [
+    { "channel_id": "0xabc123", "status": "opened" }
+  ], 1713870200],
+  "sig": ["sigServer"]
+}
+```
+
+---
+
+### 2. `CloseVirtualChannel`
+
+#### ➤ Request
+```json
+{
+  "req": [2, "CloseVirtualChannel", [
+    {
+      "channel_id": "0xabc123",
+      "allocations": [
+        { "participant": "0xAlice", "token_address": "0xToken1", "amount": "1500000000000000000" },
+        { "participant": "0xBob",   "token_address": "0xToken1", "amount": "500000000000000000" }
+      ]
+    }
+  ], 1713870020],
+  "cid": "0xabc123",
+  "sig": ["sigAlice", "sigBob"] // The message must be signed by the specified signers on channel creation.
+}
+```
+
+#### ⇦ Response
+```json
+{
+  "res": [2, "CloseVirtualChannel", [
+    { "channel_id": "0xabc123", "status": "closed" }
+  ], 1713870220],
+  "sig": ["sigServer"]
+}
+```
+
+---
+
+### 3. `GetConfig`
+
+#### ➤ Request
+```json
+{
+  "req": [3, "GetConfig", [], 1713870050],
+  "sig": ["sigAlice"]
+}
+```
+
+#### ⇦ Response
+```json
+{
+  "res": [3, "config", [
+    { "brokerAddress": "0xBroker" }
+  ], 1713870250],
+  "sig": ["sigServer"]
+}
+```
+
+---
+
+### 4. `CloseDirectChannel`
+
+#### ➤ Request
+```json
+{
+  "req": [4, "CloseDirectChannel", [
+    { "channel_id": "0x123channel", "funds_destination": "0xAliceWallet" }
+  ], 1713870080],
+  "sig": ["sigAlice"]
+}
+```
+
+#### ⇦ Response
+```json
+{
+  "res": [4, "CloseDirectChannel", [
+    {
+      "channel_id": "0x123channel",
+      "state_data": "0x0000000000000000000000000000000000000000000000000000000000001ec7",
+      "allocations": [
+        { "participant": "0xAliceWallet", "token_address": "0xToken1", "amount": "1500000000000000000" },
+        { "participant": "0xBroker",      "token_address": "0xToken1", "amount": "500000000000000000" }
+      ],
+      "server_signature": {
+        "v": "27",
+        "r": "0xabc...",
+        "s": "0xdef..."
+      }
+    }
+  ], 1713870280],
+  "sig": ["sigServer"]
+}
+```
+
+## The following methods are available but can be deprecated:
+
+---
+
+### 5. `ListOpenParticipants` (Can be deprecated)
+
+Lists all participants that have opened a channel with broker and available funds.
```

Comment:
Firstly I wanted to say that this should be behind RBAC-protection as this *seems* to be a confidential information, but then I have realized that it is also available at the blockchain level and can not be hidden or obscured from it.
Therefore, from security perspective, I think this method can stay